cms-contentful/src/main/java/io/commercetools/sunrise/cms/contentful/ContentfulCmsService.java

```diff
@@ -52,7 +64,7 @@ protected void onSuccess(final CDAArray result) {
 
             @Override
             protected void onFailure(final Throwable error) {
-                future.completeExceptionally(error);
+                future.completeExceptionally(new CmsServiceException(error.getMessage(), error.getCause()));
```

Comment:
What kind of error do you have in the throwable that makes you discard it and keep only the cause? In a normal situation, I would put here `error` instead of the `error.getCause()`.
If you pass the error itself, then the message would be repeated and then you can pass your own message that says something like `Could not fetch the entry` or something similar. The developer would then look further down the cause for more information.